packages/eslint-plugin/src/rules/ban-instanceof-html-element/index.js

```diff
@@ -0,0 +1,115 @@
+// @ts-check
+const { AST_NODE_TYPES } = require('@typescript-eslint/experimental-utils');
+const createRule = require('../../utils/createRule');
+
+/**
+ * @typedef {import("@typescript-eslint/types/dist/ts-estree").BinaryExpression} BinaryExpression
+ * @typedef {import("@fluentui/react-utilities/src/utils/isHTMLElement").HTMLElementConstructorName} HTMLElementConstructorName
+ *
+ */
+
+module.exports = createRule({
+  name: 'ban-instanceof-html-element',
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Ban usage of instanceof HTMLElement comparison',
+      category: 'Possible Errors',
+      recommended: 'error',
+    },
+    messages: {
+      invalidBinaryExpression: 'instanceof {{right}} should be avoided, use isHTMLElement instead.',
+    },
+    fixable: 'code',
+    schema: [],
+  },
+  defaultOptions: [],
+  create: context => ({
+    /**
+     * @param {BinaryExpression} binaryExpression
+     */
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    BinaryExpression(binaryExpression) {
+      if (
+        binaryExpression.operator === 'instanceof' &&
+        binaryExpression.right.type === AST_NODE_TYPES.Identifier &&
+        constructorNamesSet.has(binaryExpression.right.name)
+      ) {
+        context.report({
+          node: binaryExpression,
+          messageId: 'invalidBinaryExpression',
+          data: {
+            right: binaryExpression.right.name,
+          },
+        });
+      }
+    },
+  }),
+});
+
+/** @type {HTMLElementConstructorName[]} */
+const constructorNames = [
```

Comment:
Not blocking. Is it possible for this to work with [custom elements](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements)?

E.g.,

```js
class MyCustomElement extends HTMLElement {
....
}
customElements.defint('my-custom-element', MyCustomElement);
const custom = document.createElement('my-custom-element');

if (custom instanceof MyCustomElement) { ...
```

My guess is that it's probably not simple and easy but on the off chance it is it would be a nice feature :)

Reply:
If we're extending `HTMLElementTagNameMap`, like [lit](https://lit.dev/docs/tools/publishing/#publish-typescript-typings) does,  or some other similar mechanism then yes @spmonahan, this is possible! Otherwise we'd need to have a list of all custom tags.

Reply:
Cool! I'm not worried about it for now but good to know it's something we can add in the future if the need arises!